api/services/diff/diff.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package containerd.v1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "github.com/containerd/containerd/api/types/mount/mount.proto";
+import "github.com/containerd/containerd/api/types/descriptor/descriptor.proto";
+
+// Diff service creates and applies diffs
+service Diff {
+	// Extract extracts the content associated with the provided
+	// digests onto the provided mounts.
+	rpc Extract(ExtractRequest) returns (ExtractResponse);
+
+	rpc Diff(DiffRequest) returns (DiffResponse);
+}
+
+message ExtractRequest {
+	// Diff is the descriptor of the diff to be extracted
+	containerd.v1.types.Descriptor diff = 1;
+
+	repeated containerd.v1.types.Mount mounts = 2;
+}
+
+message ExtractResponse {
+	// Extracted is the descriptor for the object which was extracted.
+	// If the input was a compressed blob then the result will be
+	// the descriptor for the uncompressed blob.
+	containerd.v1.types.Descriptor extracted = 1;
+}
+
+message DiffRequest {
+	repeated containerd.v1.types.Mount lower_mounts = 1;
+	repeated containerd.v1.types.Mount upper_mounts = 2;
+
+	// MediaType is the media type descriptor for the created diff
+	// object
+	string media_type = 3;
```

Comment:
Okay, I am getting on board with this design!

You declare:

1. Here is the mounts for to build a diff from.
2. Here is the type of diff I want!
3. Here is a ref that I want you to use for the result!

The API responds:

1. Here is the diff, ready for you in the content store.